### PR TITLE
Fix npm vulnerabilities (Uplift to 1.35.x)

### PR DESCRIPTION
**Pull request by bsclifton**
Uplift of https://github.com/brave/brave-core/pull/12219
Partial uplift of https://github.com/brave/brave-core/pull/12238

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [ ] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [ ] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by bsclifton** [comment]
Cancelled the OS specific runs as we just need to see the `noplatform` result 👍

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.35.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
